diff --git a/OmniBLE/OmniBLE/PumpManagerUI/ViewModels/DeactivatePodViewModel.swift b/OmniBLE/OmniBLE/PumpManagerUI/ViewModels/DeactivatePodViewModel.swift
index 011e16d..16ca82b 100644
--- a/OmniBLE/OmniBLE/PumpManagerUI/ViewModels/DeactivatePodViewModel.swift
+++ b/OmniBLE/OmniBLE/PumpManagerUI/ViewModels/DeactivatePodViewModel.swift
@@ -51,7 +51,7 @@ class DeactivatePodViewModel: ObservableObject, Identifiable {
         var actionButtonDescription: String {
             switch self {
             case .active:
-                return LocalizedString("Deactivate Pod", comment: "Action button description for deactivate while pod still active")
+                return LocalizedString("Slide to Deactivate Pod", comment: "Action button description for deactivate while pod still active")
             case .resultError:
                 return LocalizedString("Retry", comment: "Action button description for deactivate after failed attempt")
             case .deactivating:
@@ -110,6 +110,14 @@ class DeactivatePodViewModel: ObservableObject, Identifiable {
     }
     
     @Published var state: DeactivatePodViewModelState = .active
+    public var stateNeedsDeliberateUserAcceptance : Bool {
+        switch state {
+        case .active:
+            true
+        default:
+            false
+        }
+    }
 
     var error: DeactivationError? {
         if case .resultError(let error) = self.state {
diff --git a/OmniBLE/OmniBLE/PumpManagerUI/Views/DeactivatePodView.swift b/OmniBLE/OmniBLE/PumpManagerUI/Views/DeactivatePodView.swift
index 73fa312..f5a85a1 100644
--- a/OmniBLE/OmniBLE/PumpManagerUI/Views/DeactivatePodView.swift
+++ b/OmniBLE/OmniBLE/PumpManagerUI/Views/DeactivatePodView.swift
@@ -8,6 +8,7 @@
 
 import SwiftUI
 import LoopKitUI
+import SlideButton
 
 struct DeactivatePodView: View {
     
@@ -65,15 +66,8 @@ struct DeactivatePodView: View {
                     }
                     .disabled(viewModel.state.isProcessing)
                 }
-                Button(action: {
-                    viewModel.continueButtonTapped()
-                }) {
-                    Text(viewModel.state.actionButtonDescription)
-                        .accessibility(identifier: "button_next_action")
-                        .accessibility(label: Text(viewModel.state.actionButtonAccessibilityLabel))
-                        .actionButtonStyle(viewModel.state.actionButtonStyle)
-                }
-                .disabled(viewModel.state.isProcessing)
+                actionButton
+                    .disabled(viewModel.state.isProcessing)
             }
             .padding()
         }
@@ -94,4 +88,30 @@ struct DeactivatePodView: View {
             secondaryButton: .default(FrameworkLocalText("Continue", comment: "Title of button to continue discard"), action: { viewModel.discardPod() })
         )
     }
+    
+    var actionText: some View {
+        Text(self.viewModel.state.actionButtonDescription)
+            .accessibility(identifier: "button_next_action")
+            .accessibility(label: Text(self.viewModel.state.actionButtonAccessibilityLabel))
+            .font(.headline)
+    }
+    
+    @ViewBuilder
+    var actionButton: some View {
+        if self.viewModel.stateNeedsDeliberateUserAcceptance {
+            SlideButton(styling: .init(indicatorSize: 60, indicatorColor: Color.red), action: {
+                self.viewModel.continueButtonTapped()
+            }) {
+                actionText
+            }
+        } else {
+            Button(action: {
+                self.viewModel.continueButtonTapped()
+            }) {
+                actionText
+                    .actionButtonStyle(.primary)
+            }
+        }
+    }
+
 }
